src/main/drivers/compass/compass_hmc5883l.c

```diff
@@ -145,80 +153,121 @@ static void hmc5883_extiHandler(extiCallbackRec_t* cb)
 }
 #endif
 
-static void hmc5883lConfigureDataReadyInterruptHandling(void)
+static void hmc5883lConfigureDataReadyInterruptHandling(magDev_t* mag)
 {
 #ifdef USE_MAG_DATA_READY_SIGNAL
-
-    if (!(hmc5883InterruptIO)) {
+    if (mag->magIntExtiTag == IO_TAG_NONE) {
         return;
     }
+
+    const IO_t magIntIO = IOGetByTag(mag->magIntExtiTag);
+
 #ifdef ENSURE_MAG_DATA_READY_IS_HIGH
-    uint8_t status = IORead(hmc5883InterruptIO);
+    uint8_t status = IORead(magIntIO);
     if (!status) {
         return;
     }
 #endif
 
-    EXTIHandlerInit(&hmc5883_extiCallbackRec, hmc5883_extiHandler);
-    EXTIConfig(hmc5883InterruptIO, &hmc5883_extiCallbackRec, NVIC_PRIO_MAG_INT_EXTI, EXTI_Trigger_Rising);
-    EXTIEnable(hmc5883InterruptIO, true);
+#if defined (STM32F7)
+    IOInit(magIntIO, OWNER_COMPASS_EXTI, 0);
+    EXTIHandlerInit(&mag->exti, hmc5883_extiHandler);
+    EXTIConfig(magIntIO, &gmag->exti, NVIC_PRIO_MPU_INT_EXTI, IO_CONFIG(GPIO_MODE_INPUT,0,GPIO_NOPULL));
+#else
+    IOInit(magIntIO, OWNER_COMPASS_EXTI, 0);
+    IOConfigGPIO(magIntIO, IOCFG_IN_FLOATING);
+    EXTIHandlerInit(&mag->exti, hmc5883_extiHandler);
+    EXTIConfig(magIntIO, &mag->exti, NVIC_PRIO_MAG_INT_EXTI, EXTI_Trigger_Rising);
```

Comment:
Just a note:
F1,F3,F4 EXTIConfig may be eventually updated to handle IO configuration internally (to match F7)